### PR TITLE
HPCC-31753 Parquet uses field names incorrectly

### PR DESCRIPTION
**Pull request by jackdelv**
The parquet plugin doesn’t correctly use RtlFieldInfo when writing data to a file. Instead of the passing the field name from RtlFieldInfo::name around as a string it should use a pointer to the actual field info object. This was causing an error where a dataset that included fields with XPATH defined was unable to be read by the plugin with the same record definition because the xpath was not written.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by dcamper — APPROVED** [review]
Looks reasonable.

Are there regression tests anywhere for these kinds of issues?  If not, should there be?

**Comment by jackdelv** [comment]
@dcamper Currently, I am maintaining a set of regression tests in a separate branch (parquetTesting) on my forked repo. Should there be regression tests for the Parquet plugin included in the platform? Also, the intern project for building a Parquet test suite should produce a much more complete set of regression tests.

**Review by ghalliday — APPROVED** [review]
@jackdelv looks good.  One question/comment unrelated to this specific PR.  Best to address in a separate PR.

File: plugins/parquet/parquetembed.cpp [hunk 1]
```diff
@@ -1490,27 +1497,17 @@ void ParquetRowBuilder::processBeginDataset(const RtlFieldInfo *field)
  */
 void ParquetRowBuilder::processBeginRow(const RtlFieldInfo *field)
 {
-    StringBuffer xpath;
-    xpathOrName(xpath, field);
-
-    if (!xpath.isEmpty())
+    if (strncmp(field->name, "<row>", 5) != 0)
```
[the remark this answers is not in the document]
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp:1500]: I would say yes they should be the same. The inconsistency is due to me learning about new library functions as I am writing the code. My guess is that using strieq is preferred due to readability. Is that correct?

**Pull request merged**